OADP 1442 - resourceTimeouts (#58111)
openshift/openshift-docs: modules/oadp-installing-dpa.adoc

      defaultPlugins:
        - openshift <1>
        - aws
      resourceTimeout: 10m <2>
    restic:
      enable: true <3>
      podConfig:

>         nodeSelector: <node selector> <4>

Reviewer: restic:
      enable: true 
      podConfig:
        nodeSelector: <node selector> 

Specify the node selector to be supplied to Restic podSpec.

the doc is not very helpful there... please add..

something like.. "by default restic will run on all nodes, an administrator may want to constrain which nodes restic is available"

[7:49 AM](https://redhat-internal.slack.com/archives/C056PQ6AS2U/p1683121790312149)
the nodeSelector would be defined by the OpenShift Admin

  backupLocations:
    - name: default
      velero:
        provider: {provider}
        default: true
        objectStorage:
          bucket: <bucket_name> <5>
          prefix: <prefix> <6>
        config:
          region: <region>
          profile: "default"
        credential:
          key: cloud
          name: {credentials} <7>
  snapshotLocations: <8>
    - name: default
      velero:
        provider: {provider}
        config:
          region: <region> <9>
          profile: "default"
----
<1> The `openshift` plugin is mandatory.

> <2> Set how long to wait for several Velero resources before timeout occurs, such as Velero CRD availability, volumeSnapshot deletion, and backup repository availability. Default is 10m.

Reviewer 2: Missing determiner - **Default is 10m** -> **The default is 10m**

> <2> Set how long to wait for several Velero resources before timeout occurs, such as Velero CRD availability, volumeSnapshot deletion, and backup repository availability. Default is 10m.

Reviewer 3: Velero CustumeResourcceDefinition availability
maybe you mean Velero Custom Resources?
Or does it mean waiting for the Resource Definition Resource to be available? 
Also, if not here already,  maybe can a list of all Resources (of ref to velero docs)?
wdyt?

Author: @eemcmullan @kaovilai  what are your thoughts?

> <2> Set how long to wait for several Velero resources before timeout occurs, such as Velero CRD availability, volumeSnapshot deletion, and repo availability. Default is 10m.

Reviewer 4: suggestion: "repo" -> "backup repository"

> <2> Specify how many minutes to wait for several Velero resources before timeout occurs, such as Velero CRD availability, volumeSnapshot deletion, and backup repository availability. The default is 10m.

Reviewer 5: I think you pinged the wrong person.

> <2> Specify how many minutes to wait for several Velero resources before timeout occurs, such as Velero CRD availability, volumeSnapshot deletion, and backup repository availability. The default is 10m.

Author: @eemcmullan please see @amastbau comment above.

> <2> Specify how many minutes to wait for several Velero resources before timeout occurs, such as Velero CRD availability, volumeSnapshot deletion, and backup repository availability. The default is 10m.

Reviewer 6: It seems that you are still waiting on some clarification on this but here's a rewording suggestion:

```suggestion
<2> Specify the time, in minutes, to wait for Velero resources, before timeout occurs. The default is 10m.
```

Author: @sjhala-ccs this was already approved by in a different PR that the QE person is unaware of. Thank you for the very helpful review. In this particular instance, I'm going to stay with the original text.

> <2> Specify how many minutes to wait for several Velero resources before timeout occurs, such as Velero CRD availability, volumeSnapshot deletion, and backup repository availability. The default is 10m.

Reviewer: @EmilienM lolz.. GET OUT OF HERE 0/ and HI!

        provider: {provider}
----
<1> The `openshift` plugin is mandatory.

> <2> Set how long to wait for several Velero resources before timeout occurs, such as Velero CRD availability, volumeSnapshot deletion, and backup repository availability. Default is 10m.

Reviewer 2: Missing determiner **Default is 10m** -> **The default is 10m**

      defaultPlugins:
        - gcp
        - openshift <1>
      resourceTimeout: 10m <2>
    restic:
      enable: true <3>
      podConfig:
        nodeSelector: <node selector> <4>
  backupLocations:
    - velero:
        provider: {provider}
        default: true
        credential:
          key: cloud
          name: {credentials} <5>
        objectStorage:
          bucket: <bucket_name> <6>
          prefix: <prefix> <7>
  snapshotLocations: <8>
    - velero:
        provider: {provider}
        default: true
        config:
          project: <project>
          snapshotLocation: us-west1 <9>
----
<1> The `openshift` plugin is mandatory.

> <2> Set how long to wait for several Velero resources before timeout occurs, such as Velero CRD availability, volumeSnapshot deletion, and backup repository availability. Default is 10m.

Reviewer 2: Missing determiner **Default is 10m** -> **The default is 10m**

<3> Set to `false` if you want to disable the Restic installation. Restic deploys a daemon set, which means that each worker node has `Restic` pods running. You configure Restic for backups by adding `spec.defaultVolumesToRestic: true` to the `Backup` CR.
<4> Specify the node selector to be supplied to Restic podSpec.
<5> If you do not specify this value, the default name, `{credentials}`, is used. If you specify a custom name, the custom name is used for the backup location.
<6> Specify a bucket as the backup storage location. If the bucket is not a dedicated bucket for Velero backups, you must specify a prefix.
<7> Specify a prefix for Velero backups, for example, `velero`, if the bucket is used for multiple purposes.

> <8> You do not need to specify a snapshot location if you use CSI snapshots or Restic to back up PVs.

Reviewer 2: See comment line 91

      defaultPlugins:
        - aws
        - openshift <1>
      resourceTimeout: 10m <2>
    restic:
      enable: true <3>
      podConfig:
        nodeSelector: <node selector> <4>
  backupLocations:
    - velero:
        config:
          profile: "default"
          region: minio
          s3Url: <url> <5>
          insecureSkipTLSVerify: "true"
          s3ForcePathStyle: "true"
        provider: {provider}
        default: true
        credential:
          key: cloud
          name: {credentials} <6>
        objectStorage:
          bucket: <bucket_name> <7>
          prefix: <prefix> <8>
----
<1> The `openshift` plugin is mandatory.

> <2> Set how long to wait for several Velero resources before timeout occurs, such as Velero CRD availability, volumeSnapshot deletion, and backup repository availability. Default is 10m.

Reviewer 2: Missing determiner **Default is 10m** -> **The default is 10m**

        - gcp <2>
        - csi <3>
        - openshift <4>
      resourceTimeout: 10m <5>
    restic:
      enable: true <6>
      podConfig:
        nodeSelector: <node selector> <7>
  backupLocations:
    - velero:
        provider: {provider} <8>
        default: true
        credential:
          key: cloud
          name: <default_secret> <9>
        objectStorage:
          bucket: <bucket_name> <10>
          prefix: <prefix> <11>
----
<1> Optional: The `kubevirt` plugin is used with {VirtProductName}.
<2> Specify the default plugin for the backup provider, for example, `gcp`, if appropriate.
<3> Specify the `csi` default plugin if you use CSI snapshots to back up PVs. The `csi` plugin uses the link:https://{velero-domain}/docs/main/csi/[Velero CSI beta snapshot APIs]. You do not need to configure a snapshot location.
<4> The `openshift` plugin is mandatory.

> <5> Set how long to wait for several Velero resources before timeout occurs, such as Velero CRD availability, volumeSnapshot deletion, and backup repository availability. Default is 10m.

Reviewer 2: Missing determiner **Default is 10m** -> **The default is 10m**

<6> Set to `false` if you want to disable the Restic installation. Restic deploys a daemon set, which means that each worker node has `Restic` pods running. You configure Restic for backups by adding `spec.defaultVolumesToRestic: true` to the `Backup` CR.
<7> Specify the node selector to be supplied to Restic podSpec
<8> Specify the backup provider.

> <9> If you use a default plugin for the backup provider, you must specify the correct default name for the `Secret`, for example, `cloud-credentials-gcp`. If you specify a custom name, the custom name is used for the backup location. If you do not specify a `Secret` name, the default name is used.

Reviewer 2: Again there is a mismatch in sentence format, starting with transitive verb. Please, can we do something like:

Specify the correct default name for the `Secret`, for example, `cloud-credentials-gcp`, if you use a default plugin for the backup provider. If you do not specify a `Secret` name, the default name is used.

Author: There is a mismatch in sentence formatting, because in this parameter there are several options the user can follow. It is probably best to leave it as is, because it is less confusing to the reader.

Author: I followed the example below.

Reviewer 2: the example above starts with a transitive verb "**<6> Set to `false`... "** then **if you want to **. let us follow the same pattern

      podConfig:
        nodeSelector: <node_selector> <7>
  backupLocations:
    - velero:
        provider: {provider} <8>
        default: true
        credential:
          key: cloud
          name: <default_secret> <9>
        objectStorage:
          bucket: <bucket_name> <10>
          prefix: <prefix> <11>
----
<1> The `kubevirt` plugin is mandatory for {VirtProductName}.
<2> Specify the plugin for the backup provider, for example, `gcp`, if it exists.
<3> The `csi` plugin is mandatory for backing up PVs with CSI snapshots. The `csi` plugin uses the link:https://{velero-domain}/docs/main/csi/[Velero CSI beta snapshot APIs]. You do not need to configure a snapshot location.
<4> The `openshift` plugin is mandatory.
<5> Specify how many minutes to wait for several Velero resources before timeout occurs, such as Velero CRD availability, volumeSnapshot deletion, and backup repository availability. The default is 10m.
<6> Set to `false`, if you want to disable the Restic installation. Restic deploys a daemon set, which means that each worker node has `Restic` pods running. You can configure Restic for backups by adding `spec.defaultVolumesToRestic: true` to the `Backup` CR.

> <7> Specify on which nodes Restic is available. By default, Restic runs on all nodes.

Author: @sjhala-ccs Dev requested the following change to <7> the selectNodes description. Can you review and then I will squash the commits and send for Merge-review?

<8> Specify the backup provider.

> <9> If you use a default plugin for the backup provider, you must specify the correct default name for the `Secret`, for example, `cloud-credentials-gcp`. If you specify a custom name, the custom name is used for the backup location. If you do not specify a `Secret` name, the default name is used.

Reviewer 2: Transitive verb formatting. Could we try something like:

Specify the correct default name for the `Secret`, for example, `cloud-credentials-gcp`, if you use a default plugin for the backup provider. If specifying a custom name, then the custom name is used for the backup location. If you do not specify a `Secret` name, the default name is used.
